pghoard/webserver.py

```diff
@@ -345,6 +345,17 @@ def get_wal_or_timeline_file(self, site, filename, filetype):
             self._save_and_verify_restored_file(filetype, filename, prefetch_target_path, target_path)
             raise HttpResponse(status=201)
 
+        # After reaching a recovery_target and restart of a PG server, PG wants to replay and refetch
+        # files from the archive starting from the latest checkpoint. We have potentially fetched these files
+        # already earlier. Check if we have the files already and if we do, don't go over the network to refetch
+        # them yet again but just rename them to the path that PG is requesting.
+        xlog_path = os.path.join(xlog_dir, "{}".format(filename))
+        if os.path.exists(xlog_path):
+            self.server.log.debug("Requested %r, found it in pg_xlog directory as: %r, returning directly",
+                                  filename, xlog_path)
+            self._save_and_verify_restored_file(filetype, filename, xlog_path, target_path)
```

Comment:
would be good to add a comment that this method will raise ValueError if the local wal file is for example a recycled one with incorrect contents for this purpose